### ISSUE TITLE
(solved) Unable to upgrade the plugin when upgrading from Peertube 5.1.0 to 6.0.2

### ISSUE THREAD

**Issue by LecygneNoir** [JohnXLivingston/peertube-plugin-livechat#285]
Hello!

As discussed on Mastodon, here is a description of the bug found for reference.

TLDR: For those having the same bug, uninstalling the plugin, then restarting peertube, solve the problem and allow reinstallation.

**Describe the bug**

I have upgraded my peertube instance from v5.1.0 to 5.2.1 then 6.0.2

After that, I tried to upgrade the livechat plugin, which failed to register, and peertube choose to remove it. Unfortunately I am unable to find logs of this error... Peertube seems to have logs from14:00 until 19:00... Possibly something related to the fact I use docker installation and upgrade recreate containers :'(

However, the plugin was not totally removed, as there are still files in the `plugins/data` directory, and I was unable to reinstall it with an error similar to #150 :

```
2023-12-29 14:21:59.825 warn: Cannot install plugin peertube-plugin-livechat. {
   "err": {
     "stack": "Error: ENOENT: no such file or directory, open '/data/plugins/node_modules/got/node_modules/decompress-response/index.js'\n    at Object.openSync (node:fs:585:3)\n    at Object.readFileSync (node:fs:453:35)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1122:18)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/data/plugins/node_modules/got/dist/source/core/index.js:14:28)\n    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/data/plugins/node_modules/got/dist/source/as-promise/types.js:14:16)\n    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/data/plugins/node_modules/got/dist/source/as-promise/index.js:16:17)\n    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)",
     "message": "ENOENT: no such file or directory, open '/data/plugins/node_modules/got/node_modules/decompress-response/index.js'",
     "errno": -2,
     "syscall": "open",
     "code": "ENOENT",
     "path": "/data/plugins/node_modules/got/node_modules/decompress-response/index.js"
   }
}
```

Note the decompress-response is correctly present inside `/data/plugins/node_modules/decompress-response/index.js`.

I tried the solution of 150, uninstall the plugin through cli, then `rm -rf` the remaining files in `/data/plugins/data/`, but with no luck, the error was still present when trying to install from UI.

After discussing on Mastodon, I tried a full restart of the peertube instance, and the bug disappear, I am now able to install the latest (8.0.4 at the moment) version correctly.

**To Reproduce**
Not exactly sure how to reproduce.
Probably something related to upgrading Peertube then upgrading the plugin, but as I upgrade fro mv5 to v6, it's also possible than there is some breaking changes in the plugin interface sorry :-/
Also, we mentionned on Mastodon a possible issue related to Node cache inside the plugin directory.

What I can say is I do not tried to upgrade plugin **before** upgrading Peertube. I'll try this next time I upgrade Peertube to be sure, and report if needed :-)

**Expected behavior**
Correct upgrade of the plugin after upgrading peertube.

**Server (please complete the following information):**
 - OS version: Using Peertube docker image so in v5, Debian 11, with v6, Debian 12 
 - Peertube version: from v5.1.0 to 5.2.1 to 6.0.2
 - Peertube installation type: docker with official image
 - peertube-plugin-livechat version: not sure of the initial version. Installed around September 29, so I guess it was v8.0.2. Problem occurred when upgrading to v8.0.4

**Plugin diagnostic:**
Unfortunately as the first failed upgrade uninstall the plugin, I was unable to launch troubleshooting tools sorry

**Desktop (please complete the following information):**
 - OS: Fedora Linux 38
 - Browser: Firefox
 - Version: 121.0

**Additional context**
As stated before, the problem is "solved" (as in "it works now on my instance") and I open the issue for reference, help of other people if they have the same problem, and if you need me to provide more details or do some more tests :-)

Thanks a lot for your help, and the wonderful work on this plugin! 🎉

**Comment by JohnXLivingston** [comment]
Thanks for the report.

Concerning remaining files in `storage/plugins/data/peertube-plugin-livechat/` after plugin uninstall, this is expected. These files should not cause any trouble when reinstalling.

I think the issue is on the Peertube side, in the plugin dependencies installation process. But it is hard to debug. And i suspect it might be because of other installed plugins (maybe when several plugins shares dependencies, but with different package versions).
